src/common.rs

```diff
@@ -27,6 +27,7 @@ pub enum Architecture {
     S390x,
     Sbf,
     Sharc,
+    Sparc32,
```

Comment:
The official name for 32-bit SPARC is just "SPARC", so I would prefer "Sparc" here.

Also, we should later add "Sparc32Plus" to add support for the "EM_SPARC32PLUS" machine type.

Reply:
A search shows that use of "sparc32" is not uncommon, and the term "SPARC" is also used in some 64-bit contexts, so I prefer to use "Sparc32" since it is less ambiguous.

Yes, I'll accept a PR that adds support for "EM_SPARC32PLUS".